File: frontend/src/objects/ui/page/page-config.ts
import { z } from "zod";
import { PageComponentSchema } from "../component/page-component.js";
import { PageLayoutSchema } from "./page-layout.js";

export const UiEndpointSchema = z.enum(["player", "designer", "admin", "director"]);
export type UiEndpoint = z.infer<typeof UiEndpointSchema>;

export const PageSurfaceModeSchema = z.enum(["composed", "staticEmbed"]).default("composed");
export type PageSurfaceMode = z.infer<typeof PageSurfaceModeSchema>;

export const PageConfigSchema = z.object({
  id: z.string().min(1),
  name: z.string().min(1),
  path: z.string().min(1),
  roleScope: UiEndpointSchema,
  layout: PageLayoutSchema,
  surfaceMode: PageSurfaceModeSchema,
  components: z.array(PageComponentSchema),
});
export type PageConfig = z.infer<typeof PageConfigSchema>;

File: backend/microservice/src/ui/objects/page/PageConfig.scala
package microservice.ui.objects.page

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}
import microservice.ui.objects.endpoint.UiEndpoint
import microservice.ui.objects.component.PageComponent

/** 动态页面完整配置领域对象。 */
final case class PageConfig(
  id: String,
  name: String,
  path: String,
  roleScope: UiEndpoint,
  layout: PageLayout,
  components: List[PageComponent]
)

/** PageConfig 的 Circe 编解码。 */
private[ui] object PageConfig {
  implicit val encoder: Encoder[PageConfig] = deriveEncoder
  implicit val decoder: Decoder[PageConfig] = deriveDecoder
}
